guide.js

```diff
@@ -10,7 +10,29 @@ function LinkedElement(element, classname, id){
 
 function init(){
 	loadSettingsFromCookie();
-	checkIframeSize();
+	//initialize sidebar container, which we will hide situationally but only need to build once.
+	let sidebarPanel = document.getElementById("sidebar");
+	let sidebarContent = document.getElementById("sidebarContent");
+	if(sidebarPanel != null && sidebarContent != null) { //if there is a sidebarPanel and sidebarContent to populate.
+		//Build an iframeContainer to add inside of the sidebarContent
+		sidebarPanel.style.display = ""; //make the sidebar visible if it isn't already.
+		let iframeContainer = document.createElement("div");
+		iframeContainer.classList.add("iframeContainer");
+		iframeContainer.id = "iframeContainer";
+
+		var myframe = document.createElement("iframe");
+		myframe.name="myframe";
+		myframe.id="myframe";
+		myframe.classList.add("iframe");
+		myframe.src="help.html";
+		
+		//Place the myframe in the iframeContainer, then place the whole payload into the sidebarContent
+		iframeContainer.appendChild(myframe);
+		sidebarContent.appendChild(iframeContainer);
+		//Run an update to the page now that things are initialized.
+		checkIframeSize(); 
```

Comment:
reviewing this code is a TODO on my end. fine for now i think.

Reply:
It's been awhile since I've looked at the original solution that was on the master, but iirc the main difference is you had the checkIframeSize function create/destroy the entire sidebar panel section each time, whereas my solution just sets "display: none" on the side panel so it can be quicker to show/hide everything. That meant I didn't need the iframecontainer to be built repeatedly and I could have it in the init function instead. I tried to comment things as best as I could both for my own sake and for review. Lmk if you need justification for other changes I made.